components/builder-worker/src/config.rs

```diff
@@ -89,6 +95,31 @@ impl Default for JobSrvAddr {
     }
 }
 
+// Ideally we'd bake this validation into a new type to used by work_poll_interval_secs.
+// Since the value is only intended to be changed as part of development work via the
+// config file and we hope to remove the zmq use in the future, we'll do the validation
+// here in the deserialization for now. If we end up expanding on the use of the poll
+// interval, it may be worth revisiting this to make it more robust
+fn deserialize_work_poll_interval<'de, D>(duration: D) -> Result<Duration, D::Error>
+    where D: Deserializer<'de>
+{
+    let duration: u64 = match serde::Deserialize::deserialize(duration)? {
+        0 => {
+            warn!("WorkerPollInterval is 0 seconds; zmq::poll will return immediately");
+            warn!("Setting to 1 second. Trust us, your cpu fans will thank you.");
+            1
+        }
+        d @ 1..=60 => d,
+        d => {
+            warn!("WorkerPollInterval is {} seconds; This may adversely impact job throughput",
```

Comment:
This is a short term fix, but if it weren't I explicitly tell them the interval is too long.